presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -224,6 +226,32 @@ static DecodedBlockNode decodeBlock(BlockFlattener flattener, Closer blockLeaseC
         return new DecodedBlockNode(decodedBlock, ImmutableList.of());
     }
 
+    static int[] ensureCapacityWithAllocator(int[] buffer, int capacity, ExpansionFactor expansionFactor, Arrays.ExpansionOption expansionOption, ArrayAllocator allocator)
```

Comment:
Any reason not to put this method into Arrays class (and drop WithAllocator suffix)?

- static import Arrays.ExpansionOption